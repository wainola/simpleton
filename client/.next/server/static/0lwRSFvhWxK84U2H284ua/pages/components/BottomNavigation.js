module.exports=function(e){var t=require("../../../../ssr-module-cache.js");function r(n){if(t[n])return t[n].exports;var o=t[n]={i:n,l:!1,exports:{}},u=!0;try{e[n].call(o.exports,o,o.exports,r),u=!1}finally{u&&delete t[n]}return o.l=!0,o.exports}return r.m=e,r.c=t,r.d=function(e,t,n){r.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},r.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},r.t=function(e,t){if(1&t&&(e=r(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var n=Object.create(null);if(r.r(n),Object.defineProperty(n,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var o in e)r.d(n,o,function(t){return e[t]}.bind(null,o));return n},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,"a",t),t},r.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},r.p="",r(r.s=1)}({"/+P4":function(e,t,r){var n=r("Bhuq"),o=r("TRZx");function u(t){return e.exports=u=o?n:function(e){return e.__proto__||n(e)},u(t)}e.exports=u},"/+oN":function(e,t){e.exports=require("core-js/library/fn/object/get-prototype-of")},"/HRN":function(e,t){e.exports=function(e,t){if(!(e instanceof t))throw new TypeError("Cannot call a class as a function")}},"0Bsm":function(e,t,r){"use strict";var n=r("KI45"),o=n(r("UXZV")),u=n(r("/HRN")),i=n(r("WaGi")),c=n(r("ZDA2")),a=n(r("/+P4")),f=n(r("N9n2")),s=function(e){if(e&&e.__esModule)return e;var t={};if(null!=e)for(var r in e)Object.hasOwnProperty.call(e,r)&&(t[r]=e[r]);return t.default=e,t},l=function(e){return e&&e.__esModule?e:{default:e}};Object.defineProperty(t,"__esModule",{value:!0});var p=s(r("cDcd")),y=l(r("rf6O")),d=l(r("2mql")),b=r("p8BD");t.default=function(e){var t=b.getDisplayName(e),r=function(t){function r(){return(0,u.default)(this,r),(0,c.default)(this,(0,a.default)(r).apply(this,arguments))}return(0,f.default)(r,t),(0,i.default)(r,[{key:"render",value:function(){return p.default.createElement(e,(0,o.default)({router:this.context.router},this.props))}}]),r}(p.Component);return r.contextTypes={router:y.default.object},r.displayName="withRouter(".concat(t,")"),d.default(r,e)}},1:function(e,t,r){e.exports=r("6Md7")},"20a2":function(e,t,r){e.exports=r("nOHt")},"2mql":function(e,t,r){"use strict";var n=r("UWCm"),o={childContextTypes:!0,contextType:!0,contextTypes:!0,defaultProps:!0,displayName:!0,getDefaultProps:!0,getDerivedStateFromError:!0,getDerivedStateFromProps:!0,mixins:!0,propTypes:!0,type:!0},u={name:!0,length:!0,prototype:!0,caller:!0,callee:!0,arguments:!0,arity:!0},i={};i[n.ForwardRef]={$$typeof:!0,render:!0};var c=Object.defineProperty,a=Object.getOwnPropertyNames,f=Object.getOwnPropertySymbols,s=Object.getOwnPropertyDescriptor,l=Object.getPrototypeOf,p=Object.prototype;e.exports=function e(t,r,n){if("string"!=typeof r){if(p){var y=l(r);y&&y!==p&&e(t,y,n)}var d=a(r);f&&(d=d.concat(f(r)));for(var b=i[t.$$typeof]||o,h=i[r.$$typeof]||o,v=0;v<d.length;++v){var x=d[v];if(!(u[x]||n&&n[x]||h&&h[x]||b&&b[x])){var m=s(r,x);try{c(t,x,m)}catch(e){}}}return t}return t}},"6Md7":function(e,t,r){"use strict";r.r(t);var n=r("cDcd"),o=r.n(n),u=r("20a2"),i=r.n(u),c=r("KKbo");t.default=function(e){var t=e.routes;return o.a.createElement(c.Paper,null,o.a.createElement(c.BottomNavigation,{showLabels:!0},t.map(function(e){return o.a.createElement(c.BottomNavigationAction,{label:e.label,icon:o.a.createElement(e.icon,null),onClick:function(){return i.a.push(e.route)}})})))}},Bhuq:function(e,t,r){e.exports=r("/+oN")},K47E:function(e,t){e.exports=function(e){if(void 0===e)throw new ReferenceError("this hasn't been initialised - super() hasn't been called");return e}},KI45:function(e,t){e.exports=function(e){return e&&e.__esModule?e:{default:e}}},KKbo:function(e,t){e.exports=require("@material-ui/core")},N9n2:function(e,t,r){var n=r("SqZg"),o=r("vjea");e.exports=function(e,t){if("function"!=typeof t&&null!==t)throw new TypeError("Super expression must either be null or a function");e.prototype=n(t&&t.prototype,{constructor:{value:e,writable:!0,configurable:!0}}),t&&o(e,t)}},SqZg:function(e,t,r){e.exports=r("o5io")},TRZx:function(e,t,r){e.exports=r("Wk4r")},TUA0:function(e,t){e.exports=require("core-js/library/fn/object/define-property")},UWCm:function(e,t){e.exports=require("react-is")},UXZV:function(e,t,r){e.exports=r("dGr4")},WaGi:function(e,t,r){var n=r("hfKm");function o(e,t){for(var r=0;r<t.length;r++){var o=t[r];o.enumerable=o.enumerable||!1,o.configurable=!0,"value"in o&&(o.writable=!0),n(e,o.key,o)}}e.exports=function(e,t,r){return t&&o(e.prototype,t),r&&o(e,r),e}},Wk4r:function(e,t){e.exports=require("core-js/library/fn/object/set-prototype-of")},XVgq:function(e,t,r){e.exports=r("gHn/")},Z7t5:function(e,t,r){e.exports=r("vqFK")},ZDA2:function(e,t,r){var n=r("iZP3"),o=r("K47E");e.exports=function(e,t){return!t||"object"!==n(t)&&"function"!=typeof t?o(e):t}},aAV7:function(e,t){e.exports=require("core-js/library/fn/reflect/construct")},b3CU:function(e,t,r){var n=r("pbKT"),o=r("vjea");function u(t,r,i){return!function(){if("undefined"==typeof Reflect||!n)return!1;if(n.sham)return!1;if("function"==typeof Proxy)return!0;try{return Date.prototype.toString.call(n(Date,[],function(){})),!0}catch(e){return!1}}()?e.exports=u=function(e,t,r){var n=[null];n.push.apply(n,t);var u=new(Function.bind.apply(e,n));return r&&o(u,r.prototype),u}:e.exports=u=n,u.apply(null,arguments)}e.exports=u},cDcd:function(e,t){e.exports=require("react")},dGr4:function(e,t){e.exports=require("core-js/library/fn/object/assign")},"gHn/":function(e,t){e.exports=require("core-js/library/fn/symbol/iterator")},hfKm:function(e,t,r){e.exports=r("TUA0")},iZP3:function(e,t,r){var n=r("XVgq"),o=r("Z7t5");function u(e){return(u="function"==typeof o&&"symbol"==typeof n?function(e){return typeof e}:function(e){return e&&"function"==typeof o&&e.constructor===o&&e!==o.prototype?"symbol":typeof e})(e)}function i(t){return"function"==typeof o&&"symbol"===u(n)?e.exports=i=function(e){return u(e)}:e.exports=i=function(e){return e&&"function"==typeof o&&e.constructor===o&&e!==o.prototype?"symbol":u(e)},i(t)}e.exports=i},nOHt:function(e,t,r){"use strict";var n=r("KI45"),o=n(r("UXZV")),u=n(r("iZP3")),i=n(r("b3CU")),c=n(r("hfKm")),a=function(e){return e&&e.__esModule?e:{default:e}};Object.defineProperty(t,"__esModule",{value:!0});var f=a(r("qxCs")),s={router:null,readyCallbacks:[],ready:function(e){if(this.router)return e();"undefined"!=typeof window&&this.readyCallbacks.push(e)}},l=["pathname","route","query","asPath"],p=["components"],y=["push","replace","reload","back","prefetch","beforePopState"];function d(){if(!s.router){throw new Error('No router instance found.\nYou should only use "next/router" inside the client side of your app.\n')}}Object.defineProperty(s,"events",{get:function(){return f.default.events}}),p.concat(l).forEach(function(e){(0,c.default)(s,e,{get:function(){return d(),s.router[e]}})}),y.forEach(function(e){s[e]=function(){var t;return d(),(t=s.router)[e].apply(t,arguments)}}),["routeChangeStart","beforeHistoryChange","routeChangeComplete","routeChangeError","hashChangeStart","hashChangeComplete"].forEach(function(e){s.ready(function(){f.default.events.on(e,function(){var t="on".concat(e.charAt(0).toUpperCase()).concat(e.substring(1));if(s[t])try{s[t].apply(s,arguments)}catch(e){console.error("Error when running the Router event: ".concat(t)),console.error("".concat(e.message,"\n").concat(e.stack))}})})}),t.default=s;var b=r("0Bsm");t.withRouter=b.default,t.createRouter=function(){for(var e=arguments.length,t=new Array(e),r=0;r<e;r++)t[r]=arguments[r];return s.router=(0,i.default)(f.default,t),s.readyCallbacks.forEach(function(e){return e()}),s.readyCallbacks=[],s.router},t.Router=f.default,t.makePublicRouterInstance=function(e){for(var t={},r=0;r<l.length;r++){var n=l[r];"object"!==(0,u.default)(e[n])?t[n]=e[n]:t[n]=(0,o.default)({},e[n])}return t.events=f.default.events,p.forEach(function(r){(0,c.default)(t,r,{get:function(){return e[r]}})}),y.forEach(function(r){t[r]=function(){return e[r].apply(e,arguments)}}),t}},o5io:function(e,t){e.exports=require("core-js/library/fn/object/create")},p8BD:function(e,t){e.exports=require("next-server/dist/lib/utils")},pbKT:function(e,t,r){e.exports=r("aAV7")},qxCs:function(e,t){e.exports=require("next-server/dist/lib/router/router")},rf6O:function(e,t){e.exports=require("prop-types")},vjea:function(e,t,r){var n=r("TRZx");function o(t,r){return e.exports=o=n||function(e,t){return e.__proto__=t,e},o(t,r)}e.exports=o},vqFK:function(e,t){e.exports=require("core-js/library/fn/symbol")}});