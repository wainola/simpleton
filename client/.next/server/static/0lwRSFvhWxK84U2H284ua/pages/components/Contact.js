module.exports=function(e){var t=require("../../../../ssr-module-cache.js");function n(r){if(t[r])return t[r].exports;var o=t[r]={i:r,l:!1,exports:{}},a=!0;try{e[r].call(o.exports,o,o.exports,n),a=!1}finally{a&&delete t[r]}return o.l=!0,o.exports}return n.m=e,n.c=t,n.d=function(e,t,r){n.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},n.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},n.t=function(e,t){if(1&t&&(e=n(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(n.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var o in e)n.d(r,o,function(t){return e[t]}.bind(null,o));return r},n.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return n.d(t,"a",t),t},n.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},n.p="",n(n.s=2)}({2:function(e,t,n){e.exports=n("ROfE")},"4mXO":function(e,t,n){e.exports=n("k1wZ")},"9Pu4":function(e,t){e.exports=require("@material-ui/core/styles")},BwKw:function(e,t,n){"use strict";n.r(t);var r=n("cDcd"),o=n.n(r);t.default=function(e){var t=e.isComplete;return o.a.createElement("div",{className:"circle-loader ".concat(t?"load-complete":"").trim()},o.a.createElement("div",{className:"checkmark draw ".concat(t?"completed":"").trim()}))}},"J3/a":function(e,t){e.exports=require("core-js/library/fn/get-iterator")},"Jo+v":function(e,t,n){e.exports=n("Z6Kq")},KKbo:function(e,t){e.exports=require("@material-ui/core")},R2Q7:function(e,t){e.exports=require("core-js/library/fn/array/is-array")},ROfE:function(e,t,n){"use strict";n.r(t);var r=n("hfKm"),o=n.n(r);function a(e,t,n){return t in e?o()(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}var i=n("Jo+v"),l=n.n(i),c=n("4mXO"),u=n.n(c),m=n("pLtp"),s=n.n(m);function f(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{},r=s()(n);"function"==typeof u.a&&(r=r.concat(u()(n).filter(function(e){return l()(n,e).enumerable}))),r.forEach(function(t){a(e,t,n[t])})}return e}var d=n("p0XB"),p=n.n(d);var b=n("XXOK"),x=n.n(b);function y(e,t){return function(e){if(p()(e))return e}(e)||function(e,t){var n=[],r=!0,o=!1,a=void 0;try{for(var i,l=x()(e);!(r=(i=l.next()).done)&&(n.push(i.value),!t||n.length!==t);r=!0);}catch(e){o=!0,a=e}finally{try{r||null==l.return||l.return()}finally{if(o)throw a}}return n}(e,t)||function(){throw new TypeError("Invalid attempt to destructure non-iterable instance")}()}var v=n("cDcd"),E=n.n(v),F=n("9Pu4"),g=n("KKbo"),h=n("BwKw");t.default=Object(F.withStyles)(function(e){return{paper:{textAlign:"center",height:"100%"},formFields:{margin:"10px",width:"75%"},formContainer:{display:"flex","flex-direction":"column","justify-content":"center","align-items":"center"},inlineFields:{display:"flex","flex-direction":"row","align-items":"flex-end"}}})(function(e){var t=y(E.a.useState(!1),2),n=(t[0],t[1],y(E.a.useState({nombre:!1,apellido:!1,email:!1,telefono:!1,razon:!1,direccion:!1}),2)),r=n[0],o=n[1],i=y(E.a.useState(!1),2),l=(i[0],i[1],y(E.a.useState(!1),2));function c(e){if(console.log("handleBlur",e.target),"direccion"!==e.target.name)return o(f({},r,a({},e.target.name,!r[e.target.name])));o(f({},r,a({},e.target.name,!r[e.target.name])))}function u(e){return o(f({},r,a({},e.target.name,!r[e.target.name])))}l[0],l[1];var m=e.classes;return console.log("props",e),E.a.createElement("div",null,E.a.createElement(g.Grid,{container:!0},E.a.createElement(g.Grid,{item:!0,xs:12,sm:12,md:12,lg:12,className:m.formContainer},E.a.createElement(g.FormControl,{className:m.formFields},E.a.createElement("div",{className:m.inlineFields},E.a.createElement(h.default,{isComplete:r.nombre,name:"nombre"}),E.a.createElement(g.TextField,{id:"input-with-icon-textfield",label:"Nombre",onBlur:r.nombre?function(){}:c,onFocus:u,name:"nombre"}))),E.a.createElement(g.FormControl,{className:m.formFields},E.a.createElement("div",{className:m.inlineFields},E.a.createElement(h.default,{isComplete:r.apellido,name:"apellido"}),E.a.createElement(g.TextField,{id:"input-with-icon-textfield",label:"Apellido",name:"apellido",onBlur:r.apellido?function(){}:c,onFocus:u}))),E.a.createElement(g.FormControl,{className:m.formFields},E.a.createElement("div",{className:m.inlineFields},E.a.createElement(h.default,{isComplete:r.email,name:"email"}),E.a.createElement(g.TextField,{id:"input-with-icon-textfield",label:"Email",name:"email",onBlur:r.email?function(){}:c,onFocus:u}))),E.a.createElement(g.FormControl,{className:m.formFields},E.a.createElement("div",{className:m.inlineFields},E.a.createElement(h.default,{isComplete:r.telefono,name:"telefono"}),E.a.createElement(g.TextField,{id:"input-with-icon-textfield",label:"Teléfono",name:"telefono",onBlur:r.telefono?function(){}:c,onFocus:u}))),E.a.createElement(g.FormControl,{className:m.formFields},E.a.createElement("div",{className:m.inlineFields},E.a.createElement(h.default,{isComplete:r.razon,name:"razon"}),E.a.createElement(g.TextField,{id:"input-with-icon-textfield",label:"Motivo de consulta",name:"razon",onBlur:r.razon?function(){}:c,onFocus:u}))),E.a.createElement(g.FormControl,{className:m.formFields},E.a.createElement("div",{className:m.inlineFields},E.a.createElement(h.default,{isComplete:r.direccion,name:"direccion"}),E.a.createElement(g.TextField,{id:"input-with-icon-textfield",label:"Dirección",name:"direccion",onBlur:r.direccion?function(){}:c,onFocus:u}))),E.a.createElement(g.FormControl,{className:m.formFields},E.a.createElement(g.Button,{color:"primary"},"Guardar")))))})},TUA0:function(e,t){e.exports=require("core-js/library/fn/object/define-property")},XXOK:function(e,t,n){e.exports=n("J3/a")},Z6Kq:function(e,t){e.exports=require("core-js/library/fn/object/get-own-property-descriptor")},cDcd:function(e,t){e.exports=require("react")},hfKm:function(e,t,n){e.exports=n("TUA0")},k1wZ:function(e,t){e.exports=require("core-js/library/fn/object/get-own-property-symbols")},p0XB:function(e,t,n){e.exports=n("R2Q7")},pLtp:function(e,t,n){e.exports=n("qJj/")},"qJj/":function(e,t){e.exports=require("core-js/library/fn/object/keys")}});