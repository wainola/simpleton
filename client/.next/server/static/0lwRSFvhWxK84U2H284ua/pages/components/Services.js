module.exports=function(e){var t=require("../../../../ssr-module-cache.js");function a(n){if(t[n])return t[n].exports;var r=t[n]={i:n,l:!1,exports:{}},l=!0;try{e[n].call(r.exports,r,r.exports,a),l=!1}finally{l&&delete t[n]}return r.l=!0,r.exports}return a.m=e,a.c=t,a.d=function(e,t,n){a.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},a.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},a.t=function(e,t){if(1&t&&(e=a(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var n=Object.create(null);if(a.r(n),Object.defineProperty(n,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var r in e)a.d(n,r,function(t){return e[t]}.bind(null,r));return n},a.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return a.d(t,"a",t),t},a.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},a.p="",a(a.s=4)}({"3PpT":function(e,t){e.exports=require("@material-ui/icons/ExpandMore")},4:function(e,t,a){e.exports=a("dR0L")},"9Pu4":function(e,t){e.exports=require("@material-ui/core/styles")},DRgV:function(e,t){e.exports=require("@material-ui/core/ExpansionPanel")},Hd3h:function(e,t){e.exports=require("@material-ui/core/ExpansionPanelSummary")},SnMI:function(e,t){e.exports=require("@material-ui/icons/WorkOutline")},UVoM:function(e,t){e.exports=require("@material-ui/core/Typography")},cDcd:function(e,t){e.exports=require("react")},dR0L:function(e,t,a){"use strict";a.r(t);var n=a("cDcd"),r=a.n(n),l=a("9Pu4"),i=a("DRgV"),o=a.n(i),u=a("Hd3h"),c=a.n(u),s=a("h1YJ"),m=a.n(s),p=a("UVoM"),d=a.n(p),f=a("3PpT"),x=a.n(f),E=a("yFMe"),y=a.n(E),g=a("kzs/"),h=a.n(g),b=a("SnMI"),P=a.n(b);t.default=Object(l.withStyles)(function(e){return{root:{width:"100%"},heading:{fontSize:e.typography.pxToRem(20),fontWeight:e.typography.fontWeightRegular,display:"flex"},headline:{"margin-left":"10px","margin-top":"-1px"}}})(function(e){var t=e.classes;return r.a.createElement("div",{className:t.root},r.a.createElement(o.a,null,r.a.createElement(c.a,{expandIcon:r.a.createElement(x.a,null)},r.a.createElement(d.a,{className:t.heading},r.a.createElement("span",null,r.a.createElement(y.a,null)),r.a.createElement("span",{className:t.headline},"Divorcios"))),r.a.createElement(m.a,null,r.a.createElement(d.a,null,"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."))),r.a.createElement(o.a,null,r.a.createElement(c.a,{expandIcon:r.a.createElement(x.a,null)},r.a.createElement(d.a,{className:t.heading},r.a.createElement("span",null,r.a.createElement(h.a,null)),r.a.createElement("span",{className:t.headline},"Separaci√≥n judicial"))),r.a.createElement(m.a,null,r.a.createElement(d.a,null,"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."))),r.a.createElement(o.a,null,r.a.createElement(c.a,{expandIcon:r.a.createElement(x.a,null)},r.a.createElement(d.a,{className:t.heading},r.a.createElement("span",null,r.a.createElement(P.a,null)),r.a.createElement("span",{className:t.headline},"Otros")))))})},h1YJ:function(e,t){e.exports=require("@material-ui/core/ExpansionPanelDetails")},"kzs/":function(e,t){e.exports=require("@material-ui/icons/Mood")},yFMe:function(e,t){e.exports=require("@material-ui/icons/People")}});