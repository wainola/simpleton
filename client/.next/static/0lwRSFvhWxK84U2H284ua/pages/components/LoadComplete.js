(window.webpackJsonp=window.webpackJsonp||[]).push([[10],{BwKw:function(e,t,a){"use strict";a.r(t);var n=a("q1tI"),o=a.n(n);a("7x88");t.default=function(e){var t=e.isComplete;return o.a.createElement("div",{className:"circle-loader ".concat(t?"load-complete":"").trim()},o.a.createElement("div",{className:"checkmark draw ".concat(t?"completed":"").trim()}))}},moJh:function(e,t,a){(window.__NEXT_P=window.__NEXT_P||[]).push(["/components/LoadComplete",function(){var e=a("BwKw");return{page:e.default||e}}])}},[["moJh",1,0,2]]]);