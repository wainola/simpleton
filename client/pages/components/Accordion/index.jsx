import React from 'react';

class Accordion extends React.Component {
  constructor() {
    super();
    this.state = {
      services: [
        {
          group: 'Divorcios',
          active: true,
          services: [
            {
              title: 'Divorcios',
              description:
                'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Animi, itaque.'
            },
            {
              title: 'Divorcio de mutuo acuerdo',
              description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Animi, itaque'
            },
            {
              title: 'Divorcios unilaterales',
              description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Animi, itaque'
            }
          ]
        },
        {
          group: 'Separaciones',
          active: false,
          services: [
            {
              title: 'Separaci贸n judicial',
              description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Animi, itaque'
            },
            {
              title: 'Liquidaci贸n de sociedad conyugal',
              description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Animi, itaque'
            }
          ]
        },
        {
          group: 'Otros',
          active: false,
          services: [
            {
              title: 'Regulaci贸n de alimentos',
              description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Animi, itaque'
            },
            {
              title: 'Relaci贸n directa y regular',
              description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Animi, itaque'
            },
            {
              title: 'Cuidado personal',
              description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Animi, itaque'
            },
            {
              title: 'Transacciones',
              description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Animi, itaque'
            }
          ]
        }
      ]
    };
    this.renderServices = this.renderServices.bind(this);
    this.handleToggle = this.handleToggle.bind(this);
  }

  handleToggle(evt) {
    evt.preventDefault();
    console.log('evt', evt.target);
    const { services } = this.state;
    services[evt.name].active = !services[evt.name].active;
    this.setState({
      ...this.state,
      ...services
    });
  }

  renderServices(servicesCollection, idx) {
    console.log('idx', idx);
    const { group, active, services } = servicesCollection;
    return (
      <AccordionItem
        idx={idx}
        group={group}
        active={active}
        services={services}
        Toggle={this.handleToggle}
      />
    );
  }

  render() {
    const { services } = this.state;
    console.log('accordion');
    return <div className="accordion-container">{services.map(this.renderServices)}</div>;
  }
}

const AccordionItem = ({ idx, group, active, services, Toggle }) => {
  return (
    <div className={`${active ? 'active' : 'inactive'} accordion-item`} onClick={Toggle} name={idx}>
      <span className="summary" name={idx}>
        {group}
      </span>
      <span className="folding-pannel answer">{services.answer}</span>
    </div>
  );
};

export default Accordion;
