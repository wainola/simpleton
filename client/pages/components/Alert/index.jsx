import React from 'react';
import { Alert } from 'react-bootstrap';

export const CustomAlert = ({ value, descriptor }) => (
  <Alert variant="danger">Problemas con sus datos</Alert>
);

export default CustomAlert;
