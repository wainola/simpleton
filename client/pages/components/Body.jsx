import React from 'react';

export default props => {
  console.log('Body', props);
  return <React.Fragment>{props.children.map(item => item)}</React.Fragment>;
};
