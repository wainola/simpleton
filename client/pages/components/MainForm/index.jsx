import React, { Component } from 'react';
import { Container, Row, Card, Form, Button } from 'react-bootstrap';
import Input from '../Input';
import CustomAlert from '../Alert';
import validations from '../../../Services/validators';

if (process.env.NODE_ENV !== 'test') {
  require('./styles.scss');
}

export class MainForm extends Component {
  constructor(props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.triggerAlert = this.triggerAlert.bind(this);
    this.state = {
      fields: {},
      invalidData: []
    };
  }

  triggerAlert() {
    console.log('tiggerAlert');
  }

  handleChange(evt) {
    evt.preventDefault();

    const { name, value } = evt.target;

    this.setState({
      fields: {
        ...this.state.fields,
        [name]: value
      }
    });
  }

  handleSubmit(evt) {
    evt.preventDefault();

    const { fields } = this.state;

    const validData = validations(fields).reduce((acc, item) => {
      const valid = !item.isValid && item;
      if (valid) {
        acc.push(item);
      }
      return acc;
    }, []);

    if (validData.length !== 0) {
      this.setState(
        {
          invalidData: validData
        },
        () => this.triggerAlert()
      );
    }
  }

  render() {
    return (
      <Container className="form-container">
        <Row>
          <Card>
            <Card.Title className="form-title">Contáctenos</Card.Title>
            <Card.Body>
              <CustomForm handleChange={this.handleChange} handleSubmit={this.handleSubmit} />
            </Card.Body>
          </Card>
        </Row>
      </Container>
    );
  }
}

export const CustomForm = ({ handleChange, handleSubmit }) => (
  <Form onSubmit={handleSubmit}>
    <Form.Group>
      <Input
        type="text"
        name="nombre"
        placeholder="ingrese su nombre"
        title="Nombre"
        handleChange={handleChange}
      />
      <Input
        type="text"
        name="apellido"
        placeholder="ingrese su su apellido"
        title="Apellido"
        handleChange={handleChange}
      />
      <Input
        type="email"
        name="email"
        placeholder="ingrese su correo electrónico"
        title="Correo electónico"
        handleChange={handleChange}
      />
      <Input
        type="phone"
        name="telefono"
        placeholder="ingrese su número de teléfono"
        title="Teléfono"
        handleChange={handleChange}
      />
      <Input
        type="textarea"
        name="razon"
        placeholder="ingrese el motivo de su consulta"
        title="Motivo de consulta"
        handleChange={handleChange}
      />
      <Input
        type="adddress"
        name="direccion"
        placeholder="ingrese su dirección"
        title="Dirección"
        handleChange={handleChange}
      />
    </Form.Group>
    <Form.Group>
      <Button variant="success" type="submit">
        Enviar!
      </Button>
    </Form.Group>
  </Form>
);

export default MainForm;
