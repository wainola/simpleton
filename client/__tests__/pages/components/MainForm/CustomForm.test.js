import { mount } from 'enzyme';
import { CustomForm } from '../../../../pages/components/MainForm';

describe('<CustomForm />', () => {});
