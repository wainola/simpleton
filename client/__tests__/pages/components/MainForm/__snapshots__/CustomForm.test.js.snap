// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomForm /> should render without errors 1`] = `
<CustomForm
  CustomAlert={[Function]}
  handleChange={[Function]}
  handleSubmit={[Function]}
  validFields={
    Object {
      "apellido": true,
      "direccion": true,
      "email": true,
      "nombre": true,
      "razon": true,
      "telefono": true,
    }
  }
>
  <Form
    as="form"
    inline={false}
    onSubmit={[Function]}
  >
    <form
      className=""
      onSubmit={[Function]}
    >
      <FormGroup
        as="div"
      >
        <div
          className="form-group"
        >
          <Input
            handleChange={[Function]}
            name="nombre"
            placeholder="ingrese su nombre"
            title="Nombre"
            type="text"
          >
            <span
              className="input-title"
            >
              Nombre
            </span>
            <input
              name="nombre"
              onChange={[Function]}
              placeholder="ingrese su nombre"
              type="text"
            />
          </Input>
          <Input
            handleChange={[Function]}
            name="apellido"
            placeholder="ingrese su su apellido"
            title="Apellido"
            type="text"
          >
            <span
              className="input-title"
            >
              Apellido
            </span>
            <input
              name="apellido"
              onChange={[Function]}
              placeholder="ingrese su su apellido"
              type="text"
            />
          </Input>
          <Input
            handleChange={[Function]}
            name="email"
            placeholder="ingrese su correo electrónico"
            title="Correo electónico"
            type="email"
          >
            <span
              className="input-title"
            >
              Correo electónico
            </span>
            <input
              name="email"
              onChange={[Function]}
              placeholder="ingrese su correo electrónico"
              type="email"
            />
          </Input>
          <Input
            handleChange={[Function]}
            name="telefono"
            placeholder="ingrese su número de teléfono"
            title="Teléfono"
            type="phone"
          >
            <span
              className="input-title"
            >
              Teléfono
            </span>
            <input
              name="telefono"
              onChange={[Function]}
              placeholder="ingrese su número de teléfono"
              type="phone"
            />
          </Input>
          <Input
            handleChange={[Function]}
            name="razon"
            placeholder="ingrese el motivo de su consulta"
            title="Motivo de consulta"
            type="textarea"
          >
            <span
              className="input-title"
            >
              Motivo de consulta
            </span>
            <input
              name="razon"
              onChange={[Function]}
              placeholder="ingrese el motivo de su consulta"
              type="textarea"
            />
          </Input>
          <Input
            handleChange={[Function]}
            name="direccion"
            placeholder="ingrese su dirección"
            title="Dirección"
            type="adddress"
          >
            <span
              className="input-title"
            >
              Dirección
            </span>
            <input
              name="direccion"
              onChange={[Function]}
              placeholder="ingrese su dirección"
              type="adddress"
            />
          </Input>
        </div>
      </FormGroup>
      <FormGroup
        as="div"
      >
        <div
          className="form-group"
        >
          <Button
            active={false}
            disabled={false}
            type="submit"
            variant="success"
          >
            <button
              className="btn btn-success"
              disabled={false}
              type="submit"
            >
              Enviar!
            </button>
          </Button>
        </div>
      </FormGroup>
    </form>
  </Form>
</CustomForm>
`;
